Source/WebCore/platform/graphics/cocoa/MediaPlayerPrivateWebM.mm

```diff
@@ -198,33 +198,68 @@ static bool isCopyDisplayedPixelBufferAvailable()
     return SourceBufferParserWebM::isContentTypeSupported(parameters.type);
 }
 
-void MediaPlayerPrivateWebM::load(const String& url)
+void MediaPlayerPrivateWebM::setPreload(MediaPlayer::Preload preload)
 {
-    ALWAYS_LOG(LOGIDENTIFIER);
+    ALWAYS_LOG(LOGIDENTIFIER, " - ", static_cast<int>(preload));
+    m_preload = preload;
+    if (m_assetURL.isEmpty())
+        return;
 
     auto player = m_player.get();
     if (!player)
         return;
 
+    if (m_preload >= MediaPlayer::Preload::MetaData && !m_resourceClient) {
+        if (!createResourceClient(m_assetURL)) {
+            ERROR_LOG(LOGIDENTIFIER, "could not create resource client");
+            setNetworkState(MediaPlayer::NetworkState::NetworkError);
+            setReadyState(MediaPlayer::ReadyState::HaveNothing);
+        }
+    }
+
     auto mimeType = player->contentMIMEType();
     if (mimeType.isEmpty() || !mimeTypeCache().contains(mimeType)) {
         ERROR_LOG(LOGIDENTIFIER, "mime type = ", mimeType, " not supported");
         setNetworkState(MediaPlayer::NetworkState::FormatError);
         return;
     }
 
+    if (preload > MediaPlayer::Preload::MetaData) {
+        for (auto it = m_readyForMoreSamplesMap.begin(); it != m_readyForMoreSamplesMap.end(); ++it)
+            notifyClientWhenReadyForMoreSamples(it->first);
+    }
+}
+
+void MediaPlayerPrivateWebM::load(const String& url)
+{
+    ALWAYS_LOG(LOGIDENTIFIER);
+
+    setNetworkState(m_preload == MediaPlayer::Preload::None ? MediaPlayer::NetworkState::Idle : MediaPlayer::NetworkState::Loading);
+    setReadyState(MediaPlayer::ReadyState::HaveNothing);
+
+    m_assetURL = URL({ }, url);
+
+    setPreload(m_preload);
```

Comment:
We trigger again preload on change of URL, but we do a !m_resourceClient which might be already created?
Do we have tests that cover this code path?

Reply:
Changing the URL in an existing MediaPlayerPrivateWebM is not going to work, even prior any of those changes.

The SourceBufferParser used to demux the webm is created at the player construction, it can't handle a change of content mid-way.

If this is to be supported, it's currently broken

Reply:
A new `MediaPlayer` is created whenever the `src` attribute is modified.